### PR TITLE
feat: code block tab

### PR DESCRIPTION
**Pull request by fiennyangeln**
## Motivation

#103 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

1. I currently changed custom-pages in the 1.0.15 version so that I can visualize how it looks like, I plan to remove it later. So, go to /docs/en/1.0.15/custom-pages:
screen shot 2018-10-24 at 11 12 02 pm
2. Try to click on the tab
3. See if the other code tab changes too or not
4. Try to resize the width of the browser to be abit smaller to see how it looks like if it overflow

## Related PRs

Null

## Todo 
1. Is the design suitable? Should we handle older browser that might not support horizontal scroll?
2. Should we have other design for smaller device?
3. Is the way I add the onclick in the js below the copy field looks right?

### CODE REVIEW

File: v1/lib/core/Doc.js [hunk 1]
```diff
@@ -59,6 +98,8 @@ class Doc extends React.Component {
       );
     }
 
+    // const a = this.props.content.split(/(DOCUSAURUS_CODE_TABS)(.*?)(END_DOCUSAURUS_CODE_TABS)/gms)
+    // console.log(a.length, a);
```
Comment (JoelMarcey) [inline v1/lib/core/Doc.js]: Remove these.

File: v1/website/versioned_docs/version-1.0.15/guides-custom-pages.md [hunk 2]
```diff
@@ -34,12 +34,108 @@ root-of-repo
 │   └── static
 ```
 
+
```
Comment (JoelMarcey) [inline v1/website/versioned_docs/version-1.0.15/guides-custom-pages.md]: Can you add this documentation to https://github.com/facebook/Docusaurus/blob/master/docs/api-doc-markdown.md instead?

File: v1/website/versioned_docs/version-1.0.15/guides-custom-pages.md [hunk 3]
```diff
@@ -34,12 +34,108 @@ root-of-repo
...
+## Codeblocks in multiple languages
+
+DOCUSAURUS_CODE_TABS
```
Comment (JoelMarcey) [inline v1/website/versioned_docs/version-1.0.15/guides-custom-pages.md]: I think this is a clever way to do this.

File: v1/website/versioned_docs/version-1.0.15/guides-custom-pages.md [hunk 4]
```diff
@@ -34,12 +34,108 @@ root-of-repo
 │   └── static
 ```
 
+
+## Codeblocks in multiple languages
+
```
Comment (JoelMarcey) [inline v1/website/versioned_docs/version-1.0.15/guides-custom-pages.md]: Let's add some wording about what is going on here.

File: v1/website/versioned_docs/version-1.0.15/guides-custom-pages.md [hunk 5]
```diff
@@ -34,12 +34,108 @@ root-of-repo
...
+END_TAB
+END_DOCUSAURUS_CODE_TABS
+
 Of course, you are also free to write your own pages. It is strongly suggested that you at least have an index page, but none of the pages provided are mandatory to include in your site. More information on how to use the provided components or include your own can be found [here](api-pages.md). Information on how to link to your different pages in the header navigation bar can be found [here](guides-navigation.md).
 
 > If you want your page to show up in your navigation header, you will need to update `siteConfig.js` to add to the `headerLinks` element. e.g., `{ page: 'about-slash', label: 'About/' }`,
 
 ## Adding Static Pages
 
+DOCUSAURUS_CODE_TABS
```
Comment (JoelMarcey) [inline v1/website/versioned_docs/version-1.0.15/guides-custom-pages.md]: I am not sure you need this all this example here if you add the documentation to https://github.com/facebook/Docusaurus/blob/master/docs/api-doc-markdown.md

File: v1/website/static/js/code-blocks-buttons.js [hunk 6]
```diff
@@ -44,4 +44,25 @@ window.addEventListener('load', function() {
       textEl.textContent = 'Copy';
     }, 2000);
   });
+
+  // add event listener for all tab
+  document.querySelectorAll('.nav-link').forEach(function(el) {
```
Comment (JoelMarcey) [inline v1/website/static/js/code-blocks-buttons.js]: We will need to make sure that this is installed as part of the Docusaurus install process -- e.g., `docusaurus-init`

**Comment by JoelMarcey** [comment]
@endiliey @yangshun - is the v1 Netlify preview broken?

**Comment by endiliey** [comment]
I think the test are failing also. Might not be because of CI

File: v1/lib/static/css/main.css [hunk 7]
```diff
@@ -2354,3 +2354,57 @@ input::placeholder {
...
+  border-bottom: 4px solid #e0e0e0;
+  width: 100%;
+  padding: 0;
+  overflow-x: auto;
+  white-space: nowrap;
+  max-height: 100%;
+}
+
+.nav-tabs::-webkit-scrollbar {
+  display: none;  // Safari and Chrome
```
Comment (fiennyangeln) [inline v1/lib/static/css/main.css]: I guess i should not put comment on this :/

Comment (yangshun) [reply v1/lib/static/css/main.css]: In CSS comments should be `/* */` not  `//`

**Comment by fiennyangeln** [comment]
Now it works, can go to https://deploy-preview-1063--docusaurus-preview.netlify.com/docs/en/1.0.15/custom-pages to see

**Comment by endiliey** [comment]
https://deploy-preview-1063--docusaurus-preview.netlify.com/docs/en/1.0.15/custom-pages

File: v1/website/static/js/code-blocks-buttons.js [hunk 8]
```diff
@@ -44,4 +44,25 @@ window.addEventListener('load', function() {
...
+      const groupId = e.target.getAttribute('data-group');
+      document
+        .querySelectorAll(`.nav-link[data-group=${groupId}]`)
+        .forEach(function(el) {
+          el.classList.remove('active');
+        });
+      document
+        .querySelectorAll(`.tab-pane[data-group=${groupId}]`)
+        .forEach(function(el) {
+          el.classList.remove('active');
```
Comment (endiliey) [inline v1/website/static/js/code-blocks-buttons.js]: This shouldnt be placed here because not every docusaurus user has this file

Comment (fiennyangeln) [reply v1/website/static/js/code-blocks-buttons.js]: Hmm yeah I'm confused where to put it actually 😯

Comment (fiennyangeln) [reply v1/website/static/js/code-blocks-buttons.js]: Should I make something like this? https://gist.github.com/yangshun/55db997ed0f8f4e6527571fc3bee4675 and put all the related css outside of the lib?

Comment (endiliey) [reply v1/website/static/js/code-blocks-buttons.js]: put it in v1/lib/static and modify the Head.js to always include it.

Comment (fiennyangeln) [reply v1/website/static/js/code-blocks-buttons.js]: @endiliey @JoelMarcey made the changes:D please review!

**Comment by endiliey** [comment]
I'm currently quite sick so my review will come late this time.

**Comment by fiennyangeln** [comment]
I will give the above a try! 💃

**Comment by JoelMarcey** [comment]
This is ready to go, I believe.

**Comment by t00f** [comment]
We tested it and are waiting for the official release 👍

**Comment by CyrilPeponnet** [comment]
Works like a charm, would be awesome to be part of next release.

**Pull request merged**